packages/@react-aria/focus/test/FocusScope.test.js

```diff
@@ -1251,4 +1251,46 @@ describe('FocusScope', function () {
       expect(document.activeElement).toBe(beforeScope);
     });
   });
+
+  describe('focusableFirstInScope', function () {
+    it('should restore focus to the first focusable or tabbable element within the scope when focus is lost within the scope', function () {
+      let {getByTestId} = render(
+        <div>
+          <FocusScope contain>
+            <div role="dialog" data-testid="focusable" tabIndex={-1}>
+              <button data-testid="tabbable1" onClick={e => act(() => e.target.remove())}>Remove me!</button>
+              <Item data-testid="item1" tabIndex={0}>Remove me, too!</Item>
+              <Item data-testid="item2" tabIndex={-1}>Remove me, three!</Item>
+            </div>
+          </FocusScope>
+        </div>
+      );
+
+      function Item(props) {
+        let focusManager = useFocusManager();
+        let onClick = e => {
+          focusManager.focusNext();
+          act(() => e.target.remove());
+        };
+        return <button {...props} tabIndex={-1} onClick={onClick} />;
```

Comment:
Is the tabIndex here supposed be before the props spread? Correct me if I'm wrong but I think you were trying to test that clicking "tabbable1" would move focus to "item1" since it is the only tabbable element in the scope?